src/main/scala/nl.knaw.dans.easy.deposit/EasyDepositApiApp.scala

```diff
@@ -70,9 +70,11 @@ class EasyDepositApiApp(configuration: Configuration) extends DebugEnhancedLoggi
     dir
   }
   private val draftsDir = getConfiguredDirectory("deposits.drafts")
-  private lazy val submitter = new Submitter(
-    stagingBaseDir = getConfiguredDirectory("deposits.stage"),
-    submitToBaseDir = getConfiguredDirectory("deposits.submit-to")
+
+  private val submitter = new Submitter(
+    getConfiguredDirectory("deposits.stage-for-submit"),
+    getConfiguredDirectory("deposits.submit-to"),
+    configuration.properties.getString("deposit.permissions.group"),
```

Comment:
I see you moved '_finding the `GroupPrinciple`_' back to `Submitter`. Why? Shouldn't an incorrect group result in a `ConfigurationException`, just as with `getConfiguredDirectory`?

Also, wouldn't moving '_finding the `GroupPrinciple`_' to here and passing a `GroupPrinciple` to `Submitter` be more beneficial for testing/mocking purposes?

Reply:
The submitter used to be lazily created by the `App` class. I didn't see the reason why and dropped the lazyness. It prevented values in `App` that were otherwise not used. Actually testing/mocking became easier.